src/process/init.rs

```diff
@@ -299,6 +299,13 @@ pub fn container_init(
         .set_id(Uid::from_raw(proc.user.uid), Gid::from_raw(proc.user.gid))
         .context("Failed to configure uid and gid")?;
 
+    // Without no new privileges, seccomp is a privileged operation. We have to
+    // do this before dropping capabilities. Otherwise, we should do it later,
+    // as close to exec as possible.
+    if linux.seccomp.is_some() && proc.no_new_privileges.is_none() {
+        seccomp::initialize_seccomp(linux.seccomp.as_ref()).context("Failed to execute seccomp")?;
+    }
```

Comment:
I think it's better not to use Option when passing to `initialize_seccomp()`. That way, you don't have to bother to check it in `initialize_seccomp()`.
```suggestion
    if proc.no_new_privileges.is_none() {
        if let Some(seccomp) = linux.seccomp.as_ref() {
            seccomp::initialize_seccomp(seccomp).context("Failed to execute seccomp")?;
        }
    }
```